python/pyarrow/_dataset.pyx

```diff
@@ -962,6 +962,10 @@ cdef class ParquetFileFragment(FileFragment):
             return None
         return [RowGroupInfo.wrap(row_group) for row_group in c_row_groups]
 
+    @property
+    def num_row_groups(self):
+        return None if self.row_groups is None else len(self.row_groups)
```

Comment:
Is there a way to do this without reifying `self.row_groups`? Or is it cheap enough?

Reply:
I'm adding a method to load just the number of row groups, leaving statistics unparsed